packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -341,13 +341,59 @@ Client](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-sett
 The following code creates an app client and retrieves the client id -
 
 ```ts
-const pool = new UserPool(this, 'Pool');
+const pool = new UserPool(this, 'pool');
+pool.addClient('customer-app-client');
+const clientId = client.userPoolClientId;
+```
 
-const client = new UserPoolClient(stack, 'Client', {
-  userPool: pool
-});
+Existing app clients can be imported into the CDK app using the `UserPoolClient.fromUserPoolClientId()` API. For new
+and imported user pools, clients can also be created via the `UserPoolClient` constructor, as so -
 
-const clientId = client.userPoolClientId; 
+```ts
+const importedPool = UserPool.fromUserPoolId(this, 'imported-pool', 'us-east-1_oiuR12Abd');
+new UserPoolClient(this, 'customer-app-client');
 ```
 
-Existing app clients can be imported into the CDK app using the `UserPoolClient.fromUserPoolClientId()` API.
+Clients can be configured with authentication flows. Authentication flows allow users on a client to be authenticated
+with a user pool. Cognito user pools provide several several different types of authentication, such as, SRP (Secure
+Remote Password) authentication, username-and-password authentication, etc. Learn more about this at [UserPool Authentication
+Flow](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-authentication-flow.html).
+
+The following code configures a client to use both SRP and username-and-password authentication -
+
+```ts
+const pool = new UserPool(this, 'pool');
+pool.addClient('app-client', {
+  authFlows: {
+    userPassword: true,
+    userSrp: true,
+  }
+});
+```
+
+Custom authentication protocols can be configured by setting the `custom` property under `authFlow` and defining lambda
```

Comment:
Kinda hate that this is implemented on 2 different resources. Do you know what the reason is?

Reply:
Assuming your question is about the CF resources, this is because each user pool client can be configured with different combination of authentication flows, of which 'custom' is one choice. So the 'custom' property is on the client.

If a 'custom' authflow is chosen, the behaviour of the backend is determined by the user via lambda functions. The backend here is represented by the user pool, hence it's configured on the user pool.